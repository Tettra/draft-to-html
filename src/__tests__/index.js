import { convertFromRaw } from 'draft-js'
import convertToHtml from '../'

describe('convertToHtml', () => {
  test('converts unstyled blocks', () => {
    const rawContent = {
      blocks: [{
        text: 'One'
      }, {
        text: 'Two'
      }],
      entityMap: {}
    }
    const html = convertToHtml(convertFromRaw(rawContent))
    expect(html).toMatchSnapshot()
  })

  test('converts unordered list', () => {
    const rawContent = {
      blocks: [{
        type: 'unordered-list-item',
        text: 'One'
      }, {
        type: 'unordered-list-item',
        text: 'Two'
      }],
      entityMap: {}
    }
    const html = convertToHtml(convertFromRaw(rawContent))
    expect(html).toMatchSnapshot()
  })

  test('converts ordered list', () => {
    const rawContent = {
      blocks: [{
        type: 'ordered-list-item',
        text: 'One'
      }, {
        type: 'ordered-list-item',
        text: 'Two'
      }],
      entityMap: {}
    }
    const html = convertToHtml(convertFromRaw(rawContent))
    expect(html).toMatchSnapshot()
  })

  test('converts adjacent unordered and ordered list', () => {
    const rawContent = {
      blocks: [{
        type: 'unordered-list-item',
        text: 'One'
      }, {
        type: 'ordered-list-item',
        text: 'Two'
      }],
      entityMap: {}
    }
    const html = convertToHtml(convertFromRaw(rawContent))
    expect(html).toMatchSnapshot()
  })

  test('converts headers', () => {
    const rawContent = {
      blocks: [{
        type: 'header-one',
        text: 'Header'
      }, {
        type: 'header-two',
        text: 'Header'
      }, {
        type: 'header-three',
        text: 'Header'
      }, {
        type: 'header-four',
        text: 'Header'
      }, {
        type: 'header-five',
        text: 'Header'
      }, {
        type: 'header-six',
        text: 'Header'
      }],
      entityMap: {}
    }
    const html = convertToHtml(convertFromRaw(rawContent))
    expect(html).toMatchSnapshot()
  })

  test('converts block quotes', () => {
    const rawContent = {
      blocks: [{
        type: 'blockquote',
        text: 'Simon says let the blockquote speak'
      }],
      entityMap: {}
    }

    const html = convertToHtml(convertFromRaw(rawContent))
    expect(html).toMatchSnapshot()
  })

  test('converts code blocks', () => {
    const rawContent = {
      blocks: [{
        type: 'code-block',
        text: 'Simon says let the code block speak'
      }],
      entityMap: {}
    }

    const html = convertToHtml(convertFromRaw(rawContent))
    expect(html).toMatchSnapshot()
  })

  test('converts inline styles', () => {
    const rawContent = {
      blocks: [{
        text: 'Simon says let the code block speak',
        inlineStyleRanges: [
          { style: 'bold', offset: 0, length: 8 }
        ],
        entityRanges: [
          { key: '1', offset: 0, length: 3 }
        ]
      }],
      entityMap: {
        '1': {
          type: 'LINK',
          data: {
            src: 'https://placekitten.com/400/300'
          }
        }
      }
    }

    const html = convertToHtml(convertFromRaw(rawContent))
    expect(html).toMatchSnapshot()
  })
})
