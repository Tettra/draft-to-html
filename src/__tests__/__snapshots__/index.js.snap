// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`convertToHtml converts adjacent unordered and ordered list 1`] = `"<ul><li>One</li></ul><ol><li>Two</li></ol>"`;

exports[`convertToHtml converts block quotes 1`] = `"<blockquote>Simon says let the blockquote speak</blockquote>"`;

exports[`convertToHtml converts code blocks 1`] = `"<pre>Simon says let the code block speak</pre>"`;

exports[`convertToHtml converts headers 1`] = `"<h1>Header</h1><h2>Header</h2><h3>Header</h3><h4>Header</h4><h5>Header</h5><h6>Header</h6>"`;

exports[`convertToHtml converts inline styles 1`] = `"<p><a href=\\"https://placekitten.com/400/300\\"><strong>Sim</strong></a><strong>on sa</strong>ys let the code block speak</p>"`;

exports[`convertToHtml converts ordered list 1`] = `"<ol><li>One</li><li>Two</li></ol>"`;

exports[`convertToHtml converts unordered list 1`] = `"<ul><li>One</li><li>Two</li></ul>"`;

exports[`convertToHtml converts unstyled blocks 1`] = `"<p>One</p><p>Two</p>"`;
